time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateTimeArbitrary.java

```diff
@@ -20,22 +22,65 @@
 	private LocalDateTime min = null;
 	private LocalDateTime max = null;
 
+	private ChronoUnit ofPrecision = DefaultLocalTimeArbitrary.DEFAULT_PRECISION;
+
 	@Override
 	protected Arbitrary<LocalDateTime> arbitrary() {
 
-		LocalDateTime effectiveMin = min != null ? min : DEFAULT_MIN;
-		LocalDateTime effectiveMax = max != null ? max : DEFAULT_MAX;
+		LocalDateTime effectiveMin = calculateEffectiveMin();
+		LocalDateTime effectiveMax = calculateEffectiveMax();
+		if (effectiveMax.isBefore(effectiveMin)) {
+			throw new IllegalArgumentException("The maximum date time is too soon after the minimum date time.");
+		}
 
 		LocalDateArbitrary dates = Dates.dates();
 		LocalTimeArbitrary times = generateTimeArbitrary(effectiveMin, effectiveMax);
 
 		dates = dates.atTheEarliest(effectiveMin.toLocalDate());
 		dates = dates.atTheLatest(effectiveMax.toLocalDate());
 
+		times = times.ofPrecision(ofPrecision);
+
 		Arbitrary<LocalDateTime> dateTimes = Combinators.combine(dates, times).as(LocalDateTime::of);
 
-		return dateTimes.filter(v -> !v.isBefore(effectiveMin) && !v.isAfter(effectiveMax));
+		dateTimes = dateTimes.filter(v -> !v.isBefore(effectiveMin) && !v.isAfter(effectiveMax))
+							 .edgeCases(edgeCases -> edgeCases.add(effectiveMin, effectiveMax));
+
+		return dateTimes;
```

Comment:
With this line (`.edgeCases(edgeCases -> edgeCases.add(effectiveMin, effectiveMax))`) shrinking no longer works.

Do you know the problem?

```
timestamp = 2021-03-12T20:05:46.612587200, Shrinking:defaultShrinking = 
  org.opentest4j.AssertionFailedError:
    expected: 1900-01-01T00:00 (java.time.LocalDateTime)
    but was : 2500-12-31T23:59:59 (java.time.LocalDateTime)
    when comparing values using 'ChronoLocalDateTime.timeLineOrder()'
```
(tries 23, checks 23)